server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,358 @@
+            "Could not open the file {}".format(list_s3))
+        return Status.FAIL
+
+    list_backup = os.path.join(tmpdir, backup_obj.list_name)
+    try:
+        with open(list_backup) as f:
+            backup_content = f.readlines()
+    except Exception:
+        logger.exception(
+            "Could not open the file {}".format(list_backup))
+        return Status.FAIL
+    else:
+        backup_content_list = []
+        for content in backup_content:
+            result_name = content.strip("\n")
+            local_result_path = os.path.join(backup_obj.dirname, result_name)
+            local_result_md5 = "{}.md5".format(local_result_path)
+            try:
+                with open(local_result_md5) as k:
+                    md5_local = k.readline().split(" ")[0]
+            except Exception:
+                logger.exception(
+                    "Could not open the file {}".format(local_result_md5))
+                continue
+            else:
+                backup_content_list.append(Entry(result_name, md5_local))
+
+    # Compare the two lists and report the differences.
+    sorted_s3_content = sorted(s3_content_list, key=lambda k: k.name)
+    sorted_backup_content = sorted(backup_content_list, key=lambda k: k.name)
+    len_s3_content = len(sorted_s3_content)
+    len_backup_content = len(sorted_backup_content)
+    i, j = 0, 0
+    while (i < len_s3_content) and (j < len_backup_content):
+        if sorted_s3_content[i] == sorted_backup_content[j]:
+            i += 1
+            j += 1
+        elif sorted_s3_content[i].name == sorted_backup_content[j].name:
+            # the md5s are different even though the names are the same
+            report_text = "Md5 check failed for: {}\n".format(
+                sorted_s3_content[i].name)
+            report.write(report_text)
+            i += 1
+            j += 1
+        elif sorted_s3_content[i].name < sorted_backup_content[j].name:
+            report_text = "{}: present in s3 but not in {}\n".format(
+                sorted_s3_content[i].name, backup_obj.description)
+            report.write(report_text)
+            i += 1
+        elif sorted_s3_content[i].name > sorted_backup_content[j].name:
+            report_text = "{}: present in {} but not in s3\n".format(
+                sorted_backup_content[j].name, backup_obj.description)
+            report.write(report_text)
+            j += 1
+
+    if i == len_s3_content and j < len_backup_content:
+        for i in sorted_backup_content[j:len_backup_content]:
+            report_text = "{}: present in {} but not in s3\n".format(
+                i.name, backup_obj.description)
+            report.write(report_text)
+    elif i < len_s3_content and j == len_backup_content:
+        for i in sorted_s3_content[i:len_s3_content]:
+            report_text = "{}: present in s3 but not in {}\n".format(
+                i.name, backup_obj.description)
+            report.write(report_text)
+    return Status.SUCCESS
```

Comment:
Lines 171 - 208: make a separate function out of this. It should take the two lists to compare as arguments. The function can then be called from this site to compare archive and s3 backup; it can also be called from  main():l321  to compare archive with the local backup.

Reply:
The report is indexed into ES. I wonder whether all of this should go into the log  and just a summary should be sent to ES instead. TBD.